### DISCUSSION TITLE
More flexibility in requests/responses

### DISCUSSION THREAD

**Discussion by keribend** [Huseyinnurbaki/mocktail#5] [Ideas]
## The Problem

As backend developer, I tried to use mocktail to mock third parties APIs responses.
Unfortunately when I try to store an XML string as response, I got the JSON.parse error notification toast. Thus, it seems that by default mocktail assumes that JSON format is used for communication.
Also, I might need to mock custom headers in HTTP requests/responses with plain text, i.e.:
- X-Operator: operator-unique-id-plus-name
- X-Another-custom-header: literally-anything-i-might-need: in any format i want!

## The proposed solution

Ideally, that would be nice to have options of flexibility in terms of requests/responses.
Something like a selector for the body format between:
- application/json
- application/xml

**Comment by Huseyinnurbaki** [comment]
Hi,

Sounds great. Let me see what I can do  👍
